### ISSUE TITLE
Refactor config properties to kebab case

### ISSUE THREAD

**Issue by nicoloboschi** [LangStream/langstream#510]
I’ve noticed different config naming styles: 90% is kebab case (my-property), the rest is pascal (myProperty) and one is with underscore (my_property). 
We should move them all to the kebab case and keep compatibility

**Comment by nicoloboschi** [comment]
Some examples:
- language-detector uses `allowedLanguages`
- unwrap-key-value uses `unwrapKey`
- text-splitter uses `splitter_type`, `length_function` and others

**Comment by eolivelli** [comment]
I agree that we should fix all of them as use the kebab case, that is more JSON/YAML friendly.

**Comment by cbornet** [comment]
💯
